### PR TITLE
Add status card eligibility to service history

### PR DESCRIPTION
**Pull request by khenson-oddball**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

- *This work is behind a feature toggle (flipper): YES*
- These changes include business logic to decide whether or not a veteran is eligible to view their proof of status card. These change includes returning a confirmed boolean and a message to display to the user if they can't view their status card.
- IIR Product team, and we own the Proof of Veteran Status feature
- Success criteria for flipper is successful backwards compatibility testing for both web and mobile /service_history endpoints.

## Related issue(s)

- [1080](https://github.com/department-of-veterans-affairs/va-iir/issues/1080)

## Testing done

- [x] *New code is covered by unit tests*
- Steps to verify this is working correctly is testing with the flipper enabled on web and mobile. After that, web and mobile will implement changes to consume the new vet_status_eligibility data.

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
This adds new data returned from the /service_history endpoint, but does not make changes to the existing data being returned. 

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Comment by jperk51** [comment]
Is the plan to have a second ticket/PR for updating mobile to include this field?

File: lib/va_profile/models/service_history.rb
Comment (AdamKing0126) [inline lib/va_profile/models/service_history.rb:96]: Not a blocker but I would recommend moving these messages out of the function and into constants on the service object.  Then they can be referenced in the tests.

File: lib/va_profile/models/service_history.rb
Comment (AdamKing0126) [inline lib/va_profile/models/service_history.rb:103]: 😎

Comment (AdamKing0126) [reply lib/va_profile/models/service_history.rb:103]: It occurs to me that maybe you might also want to move these to constants on the service object eg `HONORABLE_DISCHARGE_CODES`, `DISHONORABLE_DISCHARGE_CODES` 

Again, not a blocker.

**Review by AdamKing0126 — APPROVED** [review]
👍

File: lib/va_profile/models/service_history.rb
Comment (khenson-oddball) [reply lib/va_profile/models/service_history.rb:96]: Totally considered the constants, but decided against it because this is temporary code that will get pulled out hopefully in a couple months when we change the API. Therefore, kind of wanted to keep it contained.

**Comment by khenson-oddball** [comment]
> Is the plan to have a second ticket/PR for updating mobile to include this field?

@jperk51 do you mind providing more info? I've got the new eligibility data being returned from the /profile/service_history endpoint, which I was told is the same endpoint the mobile app uses for Proof of Status card, so believed this should set you all up to consume the data on the frontend.

**Comment by jperk51** [comment]
> > Is the plan to have a second ticket/PR for updating mobile to include this field?
> 
> @jperk51 do you mind providing more info? I've got the new eligibility data being returned from the /profile/service_history endpoint, which I was told is the same endpoint the mobile app uses for Proof of Status card, so believed this should set you all up to consume the data on the frontend.

Mobile has it's own service history endpoint which is currently used to determine showing the status card to a user. 
[Our docs](https://department-of-veterans-affairs.github.io/va-mobile-app/api/#/paths/~1v0~1military-service-history/get)
[Our controller](https://github.com/department-of-veterans-affairs/vets-api/blob/eb626f913e01036781d627c03b53d039ef9f7f4a/modules/mobile/app/controllers/mobile/v0/military_information_controller.rb#L9)

We use the same service in vets-api but not the same controller/serializer/endpoint.

**Comment by khenson-oddball** [comment]
> > > Is the plan to have a second ticket/PR for updating mobile to include this field?
> > 
> > 
> > @jperk51 do you mind providing more info? I've got the new eligibility data being returned from the /profile/service_history endpoint, which I was told is the same endpoint the mobile app uses for Proof of Status card, so believed this should set you all up to consume the data on the frontend.
> 
> Mobile has it's own service history endpoint which is currently used to determine showing the status card to a user. [Our docs](https://department-of-veterans-affairs.github.io/va-mobile-app/api/#/paths/~1v0~1military-service-history/get) [Our controller](https://github.com/department-of-veterans-affairs/vets-api/blob/eb626f913e01036781d627c03b53d039ef9f7f4a/modules/mobile/app/controllers/mobile/v0/military_information_controller.rb#L9)
> 
> We use the same service in vets-api but not the same controller/serializer/endpoint.

Apologies @jperk51, I misspoke. Looking back at my notes, it is the service that I had marked as shared between web and mobile. That's why the new data originates in that service. 

The discussions @mcommons10 and I had with mobile team folks is that there would be follow up stories we would create for the mobile team to implement the changes on the backend and frontend, pulling in this new data and using it to display either the vet status card or messages returned. Those stories have yet to be created and planned.

**Comment by jperk51** [comment]
> The discussions @mcommons10 and I had with mobile team folks is that there would be follow up stories we would create for the mobile team to implement the changes on the backend and frontend, pulling in this new data and using it to display either the vet status card or messages returned. Those stories have yet to be created and planned.

Okay cool. Sounds good. Let myself and/or @StacyB2023 know if you all need any assistance putting those tickets together.

**Comment by khenson-oddball** [comment]
> > The discussions @mcommons10 and I had with mobile team folks is that there would be follow up stories we would create for the mobile team to implement the changes on the backend and frontend, pulling in this new data and using it to display either the vet status card or messages returned. Those stories have yet to be created and planned.
> 
> Okay cool. Sounds good. Let myself and/or @StacyB2023 know if you all need any assistance putting those tickets together.

Appreciate it!

**Comment by stevenjcumming** [comment]
@khenson-oddball there's a failing testing

**Comment by khenson-oddball** [comment]
> @khenson-oddball there's a failing testing

@stevenjcumming Thanks for the alert about the tests! Thought they had passed. Got them fixed up.

**Review by AdamKing0126 — APPROVED** [review]

**Comment by RachalCassity** [comment]
Just a reminder. I assume you know to remove flippers in the console of each environment. Please remove `profile_show_proof_of_veteran_status` in the rails console, after this PR has been merged, for each environment. Thanks!

**Comment by khenson-oddball** [comment]
> ails console, after this PR has been merge

Hi @RachalCassity, thanks for the reminder. Is this something that is holding up approval of this PR?

**Comment by khenson-oddball** [comment]
> Just a reminder. I assume you know to remove flippers in the console of each environment. Please remove `profile_show_proof_of_veteran_status` in the rails console, after this PR has been merged, for each environment. Thanks!

Hi @RachalCassity, thanks for the reminder. Is this something that is holding up approval of this PR?

**Review by rmtolmach — APPROVED** [review]

**Comment by rmtolmach** [comment]
> > Just a reminder. I assume you know to remove flippers in the console of each environment. Please remove `profile_show_proof_of_veteran_status` in the rails console, after this PR has been merged, for each environment. Thanks!
> 
> Hi @RachalCassity, thanks for the reminder. Is this something that is holding up approval of this PR?

It needs to be done after the code is deployed, so no, it's not holding it up. Follow step 3 of [this section of this guide](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Removingold/unusedfeaturetoggles).

**Pull request merged**
